### PR TITLE
fix(deps): make react an optional peer dependency in `@docsearch/react`

### PR DESCRIPTION
**Pull request by brc-dd**
fixes (partially) #1272

### CODE REVIEW

**Comment by brc-dd** [comment]
Hi @shortcuts! Sorry for pinging you directly. But can this be looked into?

**Comment by benmccann** [comment]
@brc-dd it looks like this PR needs to be rebased

@shortcuts would you be able to review this PR? It's a major stumbling block to using Algolia with non-React projects like Vue, Svelte, etc. Are there any concerns you have or things we could help test?

File: packages/docsearch-react/package.json [hunk 1]
```diff
@@ -42,5 +42,16 @@
...
+  "peerDependenciesMeta": {
+    "@types/react": {
+      "optional": true
+    },
+    "react": {
+      "optional": true
+    },
+    "react-dom": {
+      "optional": true
+    }
```
Comment (shortcuts) [inline packages/docsearch-react/package.json:55]: Have you tried applying this change at the `@docsearch/js` package level? I believe it wouldn't work but I'd rather not impact this one with unrelated changes.

Ultimately, I believe it's worth the workaround to prevent warnings on non-React packages

**Review by shortcuts — COMMENTED** [review]
Thanks for the PR! I'm fine with the workaround if we can't apply it at the JS package level

File: packages/docsearch-react/package.json (hunk 1 above)
Comment (brc-dd) [reply packages/docsearch-react/package.json:55]: Yeah it won't work. `peerDependenciesMeta` can define things "related to the dependencies listed in the `peerDependencies` field". So this workaround needs to be there in the package where we have issue causing peer deps.

Comment (brc-dd) [reply packages/docsearch-react/package.json:55]: Also as you wrote at #1416, we need to make `@algolia/client-search` optional here: https://github.com/algolia/autocomplete/blob/next/packages/autocomplete-preset-algolia/package.json to fix all deps warnings (and work without error on Yarn v2/v3 and PNPM).

Comment (shortcuts) [reply packages/docsearch-react/package.json:55]: > So this workaround needs to be there in the package where we have issue causing peer deps.

I think the workaround is worth it

> Also as you wrote at #1416, we need to make `@algolia/client-search` optional here: https://github.com/algolia/autocomplete/blob/next/packages/autocomplete-preset-algolia/package.json to fix all deps warnings (and work without error on Yarn v2/v3 and PNPM).

Indeed, but actually the issue is located at the Algolia JavaScript API client level (https://github.com/algolia/algoliasearch-client-javascript/issues/1080) not re-exporting types, so Autocomplete shouldn't fix it. We've fixed it in the next major of the API client along with many other changes.

**Review by shortcuts — APPROVED** [review]
Thanks for the PR and investigation!

**Comment by benmccann** [comment]
Thanks for the merge! Any chance we can get a new release?

**Comment by shortcuts** [comment]
> Thanks for the merge! Any chance we can get a new release?

Hey, it's released in 3.2.1!

**Comment by SimenB** [comment]
@shortcuts are you sure? 😅 

https://www.npmjs.com/package/@docsearch/react
image

**Comment by shortcuts** [comment]
> @shortcuts are you sure? 😅 
> 
> 
> 
> https://www.npmjs.com/package/@docsearch/react
> 
> image
> 
> 

Oops, now it's good!
